### DISCUSSION TITLE
If there is a table or block of code,.textSelection(.enabled) only copies some (truncated by the code or table) when copying, can it be optimized?

### DISCUSSION THREAD

**Discussion by wscqs** [gonzalezreal/swift-markdown-ui#224] [Q&A] [answered]
If there is a table or block of code,.textSelection(.enabled) only copies some (truncated by the code or table) when copying, can it be optimized?

**Comment by gonzalezreal** [comment, marked as answer]
Hi @wscqs,

The `textSelection(_:)` modifier in SwiftUI selects the text on a specific `Text` view. Since a `Markdown` view has multiple `Text` views, for instance, one per paragraph, code block, or table cell, we cannot change that behavior.

If you want more control over the text that gets copied, you must implement your own copy button. You can use the styling API to access the contents of a specific block (see #215 and #220). There are more details in this discussion: #201.
